deps/rabbitmq_cli/lib/rabbitmq/cli/ctl/commands/add_vhost_command.ex

```diff
@@ -24,7 +24,19 @@ defmodule RabbitMQ.CLI.Ctl.Commands.AddVhostCommand do
     meta = %{description: desc,
              tags: parse_tags(tags),
              default_queue_type: default_qt}
-    :rabbit_misc.rpc_call(node_name, :rabbit_vhost, :add, [vhost, meta, Helpers.cli_acting_user()])
+    # check if the respective feature flag is enabled
+    case default_qt do
+      "quorum" ->
```

Comment:
does it make sense to delegate feature flag assertion and rpc call to the relevant queue_type module and have a registry of availbe queue types and do lookup/dispatch instead of hard-coded case

Reply:
`rabbit_vhost:do_add/4` can do that, we don't need a registry for that